pkg/airgap/materialize.go

```diff
@@ -0,0 +1,112 @@
+package airgap
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/replicatedhq/embedded-cluster/pkg/defaults"
+)
+
+const K0S_IMAGE_PATH = "/var/lib/k0s/images/install.tar.gz"
+
+// MaterializeAirgap places the airgap image bundle for k0s
+// this should be located at 'images-amd64.tar.gz' within embedded-cluster.tar.gz within the airgap bundle
+func MaterializeAirgap(airgapReader io.Reader) error {
+	// decompress tarball
+	ungzip, err := gzip.NewReader(airgapReader)
+	if err != nil {
+		return fmt.Errorf("failed to decompress airgap file: %w", err)
+	}
+
+	// iterate through tarball
+	tarreader := tar.NewReader(ungzip)
+	foundCharts, foundImages := false, false
+	var nextFile *tar.Header
+	for {
+		nextFile, err = tarreader.Next()
+		if err != nil {
+			if err == io.EOF {
+				return fmt.Errorf("embedded-cluster.tar.gz not found in airgap file")
+			}
+			return fmt.Errorf("failed to read airgap file: %w", err)
+		}
+
+		if nextFile.Name == "embedded-cluster/images-amd64.tar" {
+			err = writeOneFile(tarreader, K0S_IMAGE_PATH)
```

Comment:
Isn't this going to write a `tar` file with `tar.gz` suffix ?

Reply:
Good catch